src/client/runner/automation/playback/select/utils.js

```diff
@@ -0,0 +1,449 @@
+    if (!domUtils.isEditableElement(element))
+        return;
+
+    var isTextarea     = domUtils.isTextarea(element);
+    var isInputElement = domUtils.isInputElement(element);
+
+    // NOTE: we don't need to scroll input elements in Mozilla and
+    // IE > 10 because it happens automatically on selection setting
+    if (isInputElement && (browserUtils.isFirefox || browserUtils.isIE && browserUtils.version > 10))
+        return;
+
+    var elementOffset  = positionUtils.getOffsetPosition(element);
+    var elementBorders = styleUtils.getBordersWidth(element);
+    var elementScroll  = styleUtils.getElementScroll(element);
+
+    var offsetX     = point.x - elementOffset.left - elementBorders.left;
+    var offsetY     = point.y - elementOffset.top - elementBorders.top;
+    var scrollValue = null;
+
+    if (isTextarea) {
+        if (offsetY < elementScroll.top)
+            scrollValue = offsetY;
+
+        if (offsetY > element.clientHeight + elementScroll.top)
+            scrollValue = offsetY - element.clientHeight;
+
+        if (scrollValue !== null)
+            styleUtils.setScrollTop(element, Math.round(scrollValue));
+
+        return;
+    }
+
+    if (offsetX < elementScroll.left)
+        scrollValue = offsetX;
+
+    if (offsetX > element.clientWidth + elementScroll.left)
+        scrollValue = offsetX - element.clientWidth;
+
+    if (scrollValue !== null)
+        styleUtils.setScrollLeft(element, Math.round(scrollValue));
+}
+
+export function excludeElementScroll (element, point) {
+    var isTextEditable = domUtils.isTextEditableElement(element);
+    var isInputElement = domUtils.isInputElement(element);
+
+    if (!(isTextEditable || domUtils.isContentEditableElement(element)))
+        return point;
+
+    var elementOffset  = positionUtils.getOffsetPosition(element);
+    var elementBorders = styleUtils.getBordersWidth(element);
+    var elementScroll  = styleUtils.getElementScroll(element);
+    var maxLeft        = elementOffset.left + elementBorders.left + element.clientWidth;
+
+    // NOTE: we can't know input elements' scroll value in Mozilla and
+    // IE > 10 (https://bugzilla.mozilla.org/show_bug.cgi?id=293186)
+    if (isInputElement && isTextEditable &&
+        (browserUtils.isFirefox || browserUtils.isIE && browserUtils.version > 10)) {
+        return {
+            x: Math.min(point.x, maxLeft),
+            y: point.y
+        };
+    }
+
+    return {
+        x: point.x - elementScroll.left,
+        y: point.y - elementScroll.top
+    };
+}
+
+export function getNodesAndOffsetsForSetSelection (el, startPosition, endPosition, expectedStartNode, expectedEndNode) {
+    var backward       = startPosition > endPosition;
+    var startSelection = contentEditable.calculateNodeAndOffsetByPosition(el, startPosition);
+    var endSelection   = contentEditable.calculateNodeAndOffsetByPosition(el, endPosition);
+    var wrongStartNode = startSelection.node !== expectedStartNode &&
```

Comment:
This logic stored historically, I'll rewrite behavior for contenteditable elements and 'startNode', 'endNode' options